retis/src/module/skb/bpf/skb_hook.bpf.c

```diff
@@ -10,43 +10,40 @@
 #define ETH_P_ARP	0x0806
 #define ETH_P_IPV6	0x86dd
 
-/* Skb raw event sections.
- *
- * Please keep in sync with its Rust counterpart in module::skb::bpf.
- */
-#define COLLECT_PACKET		1
-#define COLLECT_DEV		2
-#define COLLECT_NS		3
-#define COLLECT_META		4
-#define COLLECT_DATA_REF	5
-#define COLLECT_GSO		6
-
-/* Skb hook configuration. A map is used to set the config from userspace.
- *
- * Please keep in sync with its Rust counterpart in module::skb::bpf.
+/* Skb raw event sections. */
+enum skb_opts {
+	SECTION_PACKET = 1,
+	SECTION_DEV,
+	SECTION_NS,
+	SECTION_META,
+	SECTION_DATA_REF,
+	SECTION_GSO,
+} __binding;
+
+/* Skb hook configuration. A map is used to set the config from
+ * userspace.
  */
 struct skb_config {
 	u64 sections;
-} __attribute__((packed));
+} __binding;
 struct {
 	__uint(type, BPF_MAP_TYPE_ARRAY);
 	__uint(max_entries, 1);
 	__type(key, u32);
 	__type(value, struct skb_config);
 } skb_config_map SEC(".maps");
 
-/* Please keep the following structs in sync with its Rust counterpart in
- * module::skb::bpf.
- */
+const u8 __binding ifnamesz = 16;
```

Comment:
Please add a comment on why we need to define this as a const variable instead of only using a define. Also we can link the two:

```c
#define IFNAMSIZ	16
const u8 __binding ifnamesz = IFNAMSIZ;
```

Reply:
This slipped out. I didn't check the other patches, but I guess they follow the pattern you suggest, or at least they are supposed to.

Reply:
Yes, I saw later patches do follow the above.